Kernel/Syscalls/chdir.cpp

```diff
@@ -41,6 +41,11 @@ ErrorOr<FlatPtr> Process::sys$getcwd(Userspace<char*> buffer, size_t size)
         return EINVAL;
 
     auto path = TRY(current_directory().try_serialize_absolute_path());
+
+    // NOTE: We use the overload of validate_path_against_process_veil that takes a StringView rather than a Custody
+    //       to avoid creating the absolute path KString twice.
+    TRY(VirtualFileSystem::the().validate_path_against_process_veil(path->view(), O_RDONLY | O_DIRECTORY));
```

Comment:
ENOENT is not a valid POSIX errno for getcwd

Reply:
If we would want to stick to [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/functions/getcwd.html), we would probably want to return `EACCES`. Although keep in mind that obviously `pledge` and `unveil` themselves are not POSIX, so we may want to deviate from POSIX here. Both [Linux](https://man7.org/linux/man-pages/man3/getcwd.3.html) and [OpenBSD](https://man.openbsd.org/getcwd.3) allow `ENOENT`, in case the working directory has been unlinked.

So there are three options:
* `EACCES` (but unveil should only return `EACCES` if the directory has been unveiled with at least one of the permissions, and `ENOENT` if it hasn't been unveiled at all).
* `ENOENT` (which would be the canonical error code according to how `unveil` works)
* Do not validate against the veil, and throw away this patch.

I don't know what would be the best idea, maybe this requires some more discussion.